llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -808,8 +808,13 @@ class EarlyCSE {
       : Inst(Inst) {
       if (IntrinsicInst *II = dyn_cast<IntrinsicInst>(Inst)) {
         IntrID = II->getIntrinsicID();
-        if (TTI.getTgtMemIntrinsic(II, Info))
+        std::pair<std::optional<MemIntrinsicInfo>,
+                  std::optional<SmallVector<InterestingMemoryOperand, 1>>>
+            MemInfo = TTI.getTgtMemIntrinsic(II);
+        if (MemInfo.first != std::nullopt) {
```

Comment:
```suggestion
        if (MemInfo.first.PtrVal) {
```
Check `PtrVal` is null or not to see whether the target fill MemIntrinsicInfo.